File: packages/sdk-go/melaya/credentials.go
// Credentials API — store, retrieve, test, and delete user-scoped secrets.
//
// Maps to /api/v1/private/credentials/*. Credentials are envelope-encrypted
// at rest. Use Connectors for project-scoped credentials.
package melaya

import (
	"context"
	"net/url"
)

// CredentialsAPI wraps user-scoped credential endpoints.
type CredentialsAPI struct {
	h *httpClient
}

// List returns all stored credentials for the caller.
//
// GET /api/v1/private/credentials
func (c *CredentialsAPI) List(ctx context.Context) ([]Credential, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials", nil)
	if err != nil {
		return nil, err
	}
	var v []Credential
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ConnectedServices lists connected third-party services.
//
// GET /api/v1/private/credentials/services
func (c *CredentialsAPI) ConnectedServices(ctx context.Context) ([]ConnectedService, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials/services", nil)
	if err != nil {
		return nil, err
	}
	var v []ConnectedService
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// Get returns a stored credential by service name. Pass key to retrieve a specific
// named key within the service.
//
// GET /api/v1/private/credentials/:service
func (c *CredentialsAPI) Get(ctx context.Context, service, key string) (*Credential, error) {
	q := map[string]string{}
	if key != "" {
		q["key"] = key
	}
	data, err := c.h.get(ctx, "/api/v1/private/credentials/"+url.PathEscape(service), q)
	if err != nil {
		return nil, err
	}
	var v Credential
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// Set stores or updates a credential (envelope-encrypted at rest).
//
// PUT /api/v1/private/credentials/:service
func (c *CredentialsAPI) Set(ctx context.Context, service string, body CredentialSetBody) (bool, error) {
	data, err := c.h.put(ctx, "/api/v1/private/credentials/"+url.PathEscape(service), body)
	if err != nil {
		return false, err
	}
	var v okResult
	if err := unmarshal(data, &v); err != nil {
		return false, err
	}
	return v.Ok, nil
}

// Delete deletes a stored credential by service name.
//
// DELETE /api/v1/private/credentials/:service
func (c *CredentialsAPI) Delete(ctx context.Context, service string) (bool, error) {
	data, err := c.h.del(ctx, "/api/v1/private/credentials/"+url.PathEscape(service), nil)
	if err != nil {
		return false, err
	}
	var v okResult
	if err := unmarshal(data, &v); err != nil {
		return false, err
	}
	return v.Ok, nil
}

// Test tests a stored credential against its target service.
//
// POST /api/v1/private/credentials/:service/test
func (c *CredentialsAPI) Test(ctx context.Context, service string) (*CredentialTestResult, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/"+url.PathEscape(service)+"/test", nil)
	if err != nil {
		return nil, err
	}
	var v CredentialTestResult
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// ── Operator profile ──────────────────────────────────────────────────────────

// GetOperatorProfile returns the operator profile (persona config for agents).
//
// GET /api/v1/private/credentials/operator-profile
func (c *CredentialsAPI) GetOperatorProfile(ctx context.Context) (*OperatorProfile, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials/operator-profile", nil)
	if err != nil {
		return nil, err
	}
	var v OperatorProfile
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// SetOperatorProfile saves the operator profile.
//
// PUT /api/v1/private/credentials/operator-profile
func (c *CredentialsAPI) SetOperatorProfile(ctx context.Context, profile OperatorProfile) (bool, error) {
	data, err := c.h.put(ctx, "/api/v1/private/credentials/operator-profile", profile)
	if err != nil {
		return false, err
	}
	var v okResult
	if err := unmarshal(data, &v); err != nil {
		return false, err
	}
	return v.Ok, nil
}

// ── AI models ─────────────────────────────────────────────────────────────────

// ListModels lists available AI models across all configured providers.
//
// GET /api/v1/private/credentials/models
func (c *CredentialsAPI) ListModels(ctx context.Context, provider, capability string) ([]AIModel, error) {
	q := map[string]string{}
	if provider != "" {
		q["provider"] = provider
	}
	if capability != "" {
		q["capability"] = capability
	}
	data, err := c.h.get(ctx, "/api/v1/private/credentials/models", q)
	if err != nil {
		return nil, err
	}
	var v []AIModel
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ── Melaya accounts ───────────────────────────────────────────────────────────

// MelayaAccounts lists Melaya sub-accounts available to the caller.
//
// GET /api/v1/private/credentials/melaya-accounts
func (c *CredentialsAPI) MelayaAccounts(ctx context.Context) ([]map[string]interface{}, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials/melaya-accounts", nil)
	if err != nil {
		return nil, err
	}
	var v []map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ── RAG ───────────────────────────────────────────────────────────────────────

// RagIngestStart starts a RAG document ingestion job.
//
// POST /api/v1/private/rag/ingest
func (c *CredentialsAPI) RagIngestStart(ctx context.Context, body RagIngestStartBody) (*RagJobStatus, error) {
	data, err := c.h.post(ctx, "/api/v1/private/rag/ingest", body)
	if err != nil {
		return nil, err
	}
	var v RagJobStatus
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// RagIngestStatus polls RAG ingestion job status.
//
// GET /api/v1/private/rag/ingest/:sessionId
func (c *CredentialsAPI) RagIngestStatus(ctx context.Context, sessionID string) (*RagJobStatus, error) {
	path := "/api/v1/private/rag/ingest/" + url.PathEscape(sessionID)
	data, err := c.h.get(ctx, path, nil)
	if err != nil {
		return nil, err
	}
	var v RagJobStatus
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// RagRetrieveStart starts a RAG retrieval query.
//
// POST /api/v1/private/rag/retrieve
func (c *CredentialsAPI) RagRetrieveStart(ctx context.Context, body RagRetrieveBody) (*RagJobStatus, error) {
	data, err := c.h.post(ctx, "/api/v1/private/rag/retrieve", body)
	if err != nil {
		return nil, err
	}
	var v RagJobStatus
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// RagRetrieveStatus polls a RAG retrieval result.
//
// GET /api/v1/private/rag/retrieve/:sessionId
func (c *CredentialsAPI) RagRetrieveStatus(ctx context.Context, sessionID string) (*RagJobStatus, error) {
	path := "/api/v1/private/rag/retrieve/" + url.PathEscape(sessionID)
	data, err := c.h.get(ctx, path, nil)
	if err != nil {
		return nil, err
	}
	var v RagJobStatus
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// ── Folder picker ─────────────────────────────────────────────────────────────

// PickFolderStart initiates a native folder picker for file ingestion.
//
// POST /api/v1/private/rag/pick-folder
func (c *CredentialsAPI) PickFolderStart(ctx context.Context) (*PickFolderStatus, error) {
	data, err := c.h.post(ctx, "/api/v1/private/rag/pick-folder", nil)
	if err != nil {
		return nil, err
	}
	var v PickFolderStatus
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// PickFolderStatus polls a folder picker result.
//
// GET /api/v1/private/rag/pick-folder/:sessionId
func (c *CredentialsAPI) PickFolderStatus(ctx context.Context, sessionID string) (*PickFolderStatus, error) {
	path := "/api/v1/private/rag/pick-folder/" + url.PathEscape(sessionID)
	data, err := c.h.get(ctx, path, nil)
	if err != nil {
		return nil, err
	}
	var v PickFolderStatus
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return &v, nil
}

// ── LinkedIn OAuth ────────────────────────────────────────────────────────────

// LinkedInConnectStart starts a LinkedIn OAuth flow.
//
// POST /api/v1/private/credentials/linkedin/connect
func (c *CredentialsAPI) LinkedInConnectStart(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/linkedin/connect", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// LinkedInConnectCancel cancels an in-progress LinkedIn OAuth flow.
//
// DELETE /api/v1/private/credentials/linkedin/connect
func (c *CredentialsAPI) LinkedInConnectCancel(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.del(ctx, "/api/v1/private/credentials/linkedin/connect", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// LinkedInConnectStatus polls the LinkedIn OAuth connection status.
//
// GET /api/v1/private/credentials/linkedin/connect/status
func (c *CredentialsAPI) LinkedInConnectStatus(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials/linkedin/connect/status", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ── Luma OAuth ───────────────────────────────────────────────────────────────

// LumaConnectStart starts a Luma OAuth flow.
//
// POST /api/v1/private/credentials/luma/connect
func (c *CredentialsAPI) LumaConnectStart(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/luma/connect", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// LumaConnectStatus polls the Luma OAuth connection status.
//
// GET /api/v1/private/credentials/luma/connect/status
func (c *CredentialsAPI) LumaConnectStatus(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials/luma/connect/status", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// LumaConnectCancel cancels an in-progress Luma OAuth flow.
//
// DELETE /api/v1/private/credentials/luma/connect
func (c *CredentialsAPI) LumaConnectCancel(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.del(ctx, "/api/v1/private/credentials/luma/connect", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// GetLumaRegistrationSchema returns the Luma event registration form schema.
//
// GET /api/v1/private/credentials/luma/schema
func (c *CredentialsAPI) GetLumaRegistrationSchema(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials/luma/schema", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ── Google OAuth ──────────────────────────────────────────────────────────────

// GoogleOAuthStart starts a Google OAuth flow for credential storage.
//
// POST /api/v1/private/credentials/google/oauth
func (c *CredentialsAPI) GoogleOAuthStart(ctx context.Context, body map[string]interface{}) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/google/oauth", body)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ── CLI auth ──────────────────────────────────────────────────────────────────

// CliAuthStart starts a CLI authentication flow (device-code style).
//
// POST /api/v1/private/credentials/cli-auth
func (c *CredentialsAPI) CliAuthStart(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/cli-auth", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ── NotebookLM ────────────────────────────────────────────────────────────────

// NotebookLMLogin stores NotebookLM credentials.
//
// POST /api/v1/private/credentials/notebooklm/login
func (c *CredentialsAPI) NotebookLMLogin(ctx context.Context, body map[string]interface{}) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/notebooklm/login", body)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// NotebookLMStatus checks NotebookLM connection status.
//
// GET /api/v1/private/credentials/notebooklm/status
func (c *CredentialsAPI) NotebookLMStatus(ctx context.Context) (map[string]interface{}, error) {
	data, err := c.h.get(ctx, "/api/v1/private/credentials/notebooklm/status", nil)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// ── Telegram auth ─────────────────────────────────────────────────────────────

// TelegramAuthStart starts Telegram user auth (phone number step).
//
// POST /api/v1/private/credentials/telegram/auth
func (c *CredentialsAPI) TelegramAuthStart(ctx context.Context, body map[string]interface{}) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/telegram/auth", body)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// TelegramAuthCode submits the Telegram SMS verification code.
//
// POST /api/v1/private/credentials/telegram/auth/code
func (c *CredentialsAPI) TelegramAuthCode(ctx context.Context, body map[string]interface{}) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/telegram/auth/code", body)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

// TelegramAuth2FA submits the Telegram 2FA password.
//
// POST /api/v1/private/credentials/telegram/auth/2fa
func (c *CredentialsAPI) TelegramAuth2FA(ctx context.Context, body map[string]interface{}) (map[string]interface{}, error) {
	data, err := c.h.post(ctx, "/api/v1/private/credentials/telegram/auth/2fa", body)
	if err != nil {
		return nil, err
	}
	var v map[string]interface{}
	if err := unmarshal(data, &v); err != nil {
		return nil, err
	}
	return v, nil
}

File: packages/sdk-rust/src/credentials.rs
use serde_json::{json, Value};
use std::collections::HashMap;

use crate::client::HttpClient;
use crate::error::Result;

/// Credentials API — store, retrieve, test, and delete secrets and third-party
/// service connections at user scope.
#[derive(Clone)]
pub struct CredentialsAPI {
    http: HttpClient,
}

impl CredentialsAPI {
    pub(crate) fn new(http: HttpClient) -> Self {
        Self { http }
    }

    /// List all stored credentials (services, OAuth connections, env handles).
    pub async fn list(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http.get("/api/v1/private/credentials", &q).await
    }

    /// List connected third-party services.
    pub async fn connected_services(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get("/api/v1/private/credentials/services", &q)
            .await
    }

    /// Get a stored credential value by service name.
    pub async fn get(&self, service: &str, key: Option<&str>) -> Result<Value> {
        let mut q: HashMap<&str, Option<String>> = HashMap::new();
        q.insert("key", key.map(str::to_owned));
        self.http
            .get(&format!("/api/v1/private/credentials/{service}"), &q)
            .await
    }

    /// Store or update a credential (envelope-encrypted at rest).
    pub async fn set(
        &self,
        service: &str,
        value: &str,
        key: Option<&str>,
        label: Option<&str>,
    ) -> Result<Value> {
        let mut body = json!({ "value": value });
        if let Some(k) = key {
            body["key"] = json!(k);
        }
        if let Some(l) = label {
            body["label"] = json!(l);
        }
        self.http
            .put(&format!("/api/v1/private/credentials/{service}"), &body)
            .await
    }

    /// Delete a stored credential by service name.
    pub async fn delete(&self, service: &str) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .delete(&format!("/api/v1/private/credentials/{service}"), &q)
            .await
    }

    /// Test a stored credential (e.g. validate an API key against its target service).
    pub async fn test(&self, service: &str) -> Result<Value> {
        self.http
            .post(
                &format!("/api/v1/private/credentials/{service}/test"),
                &json!({}),
            )
            .await
    }

    // ── Operator profile ─────────────────────────────────────────────────────

    /// Get the operator profile (persona config injected into agent context).
    pub async fn get_operator_profile(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get("/api/v1/private/credentials/operator-profile", &q)
            .await
    }

    /// Save the operator profile.
    pub async fn set_operator_profile(&self, profile: &Value) -> Result<Value> {
        self.http
            .put("/api/v1/private/credentials/operator-profile", profile)
            .await
    }

    // ── AI models ─────────────────────────────────────────────────────────────

    /// List available AI models across all configured providers.
    pub async fn list_models(
        &self,
        provider: Option<&str>,
        capability: Option<&str>,
    ) -> Result<Value> {
        let mut q: HashMap<&str, Option<String>> = HashMap::new();
        q.insert("provider", provider.map(str::to_owned));
        q.insert("capability", capability.map(str::to_owned));
        self.http
            .get("/api/v1/private/credentials/models", &q)
            .await
    }

    // ── RAG (ingestion + retrieval) ──────────────────────────────────────────

    /// Start a RAG document ingestion job.
    pub async fn rag_ingest_start(&self, body: &Value) -> Result<Value> {
        self.http.post("/api/v1/private/rag/ingest", body).await
    }

    /// Poll RAG ingestion job status.
    pub async fn rag_ingest_status(&self, session_id: &str) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get(&format!("/api/v1/private/rag/ingest/{session_id}"), &q)
            .await
    }

    /// Start a RAG retrieval query.
    pub async fn rag_retrieve_start(&self, body: &Value) -> Result<Value> {
        self.http.post("/api/v1/private/rag/retrieve", body).await
    }

    /// Poll RAG retrieval result.
    pub async fn rag_retrieve_status(&self, session_id: &str) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get(&format!("/api/v1/private/rag/retrieve/{session_id}"), &q)
            .await
    }

    // ── Folder picker ────────────────────────────────────────────────────────

    /// Initiate native folder picker for file ingestion.
    pub async fn pick_folder_start(&self) -> Result<Value> {
        self.http
            .post("/api/v1/private/rag/pick-folder", &json!({}))
            .await
    }

    /// Poll folder picker result.
    pub async fn pick_folder_status(&self, session_id: &str) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get(&format!("/api/v1/private/rag/pick-folder/{session_id}"), &q)
            .await
    }

    // ── LinkedIn OAuth ───────────────────────────────────────────────────────

    /// Start LinkedIn OAuth flow.
    pub async fn linkedin_connect_start(&self) -> Result<Value> {
        self.http
            .post("/api/v1/private/credentials/linkedin/connect", &json!({}))
            .await
    }

    /// Cancel an in-progress LinkedIn OAuth flow.
    pub async fn linkedin_connect_cancel(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .delete("/api/v1/private/credentials/linkedin/connect", &q)
            .await
    }

    /// Poll LinkedIn OAuth connection status.
    pub async fn linkedin_connect_status(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get("/api/v1/private/credentials/linkedin/connect/status", &q)
            .await
    }

    // ── Luma OAuth ──────────────────────────────────────────────────────────

    /// Start Luma OAuth flow.
    pub async fn luma_connect_start(&self) -> Result<Value> {
        self.http
            .post("/api/v1/private/credentials/luma/connect", &json!({}))
            .await
    }

    /// Poll Luma OAuth connection status.
    pub async fn luma_connect_status(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get("/api/v1/private/credentials/luma/connect/status", &q)
            .await
    }

    /// Cancel an in-progress Luma OAuth flow.
    pub async fn luma_connect_cancel(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .delete("/api/v1/private/credentials/luma/connect", &q)
            .await
    }

    /// Get the Luma event registration form schema for a given event.
    pub async fn get_luma_registration_schema(&self, event_id: Option<&str>) -> Result<Value> {
        let mut q: HashMap<&str, Option<String>> = HashMap::new();
        q.insert("eventId", event_id.map(str::to_owned));
        self.http
            .get("/api/v1/private/credentials/luma/schema", &q)
            .await
    }

    // ── Google OAuth ─────────────────────────────────────────────────────────

    /// Start Google OAuth flow for credential storage.
    pub async fn google_oauth_start(&self, body: &Value) -> Result<Value> {
        self.http
            .post("/api/v1/private/credentials/google/oauth", body)
            .await
    }

    // ── CLI auth ─────────────────────────────────────────────────────────────

    /// Start CLI authentication flow (device-code style).
    pub async fn cli_auth_start(&self) -> Result<Value> {
        self.http
            .post("/api/v1/private/credentials/cli-auth", &json!({}))
            .await
    }

    // ── NotebookLM ───────────────────────────────────────────────────────────

    /// Store NotebookLM credentials.
    pub async fn notebooklm_login(&self, body: &Value) -> Result<Value> {
        self.http
            .post("/api/v1/private/credentials/notebooklm/login", body)
            .await
    }

    /// Check NotebookLM connection status.
    pub async fn notebooklm_status(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get("/api/v1/private/credentials/notebooklm/status", &q)
            .await
    }

    // ── Telegram auth ────────────────────────────────────────────────────────

    /// Start Telegram user auth (phone number step).
    pub async fn telegram_auth_start(&self, phone: &str) -> Result<Value> {
        self.http
            .post(
                "/api/v1/private/credentials/telegram/auth",
                &json!({ "phone": phone }),
            )
            .await
    }

    /// Submit Telegram SMS verification code.
    pub async fn telegram_auth_code(&self, code: &str) -> Result<Value> {
        self.http
            .post(
                "/api/v1/private/credentials/telegram/auth/code",
                &json!({ "code": code }),
            )
            .await
    }

    /// Submit Telegram 2FA password.
    pub async fn telegram_auth_2fa(&self, password: &str) -> Result<Value> {
        self.http
            .post(
                "/api/v1/private/credentials/telegram/auth/2fa",
                &json!({ "password": password }),
            )
            .await
    }

    // ── Melaya accounts ──────────────────────────────────────────────────────

    /// List Melaya sub-accounts available to the caller.
    pub async fn melaya_accounts(&self) -> Result<Value> {
        let q = HashMap::new();
        self.http
            .get("/api/v1/private/credentials/melaya-accounts", &q)
            .await
    }
}
